rewrite-yaml/src/main/java/org/openrewrite/yaml/trait/YamlReference.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.yaml.trait;
+
+import lombok.Value;
+import org.jspecify.annotations.NonNull;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.Cursor;
+import org.openrewrite.Incubating;
+import org.openrewrite.SourceFile;
+import org.openrewrite.trait.Reference;
+import org.openrewrite.trait.SimpleTraitMatcher;
+import org.openrewrite.yaml.tree.Yaml;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+@Incubating(since = "8.40.3")
+@Value
+class YamlReference implements Reference {
```

Comment:
AFAICT this will now try to match any YAML scalar values against the regexp we have for Java identifiers. This seems wrong to me. I don't think there is any need to add this class here. Instead I would expect "domain-specific" implementations like `SpringReference` which look for type references in specific YAML documents (matching some given file name or similar).

AFAIK Spring also allows config to be in YAML, so maybe we could provide an implementation for that.